scripts/legacy_expression_tool.py

```diff
@@ -0,0 +1,394 @@
+                            find_object_occurrences.append({'_id':str(occurrence['_id']), 'expression':active['reverse']['expression'], 'type':'active.reverse.expression', 'service':occurrence['service'], 'subservice':occurrence['subservice'], 'expressionIndex':found_legacy_expressions.index(active['reverse']['expression'])})
+                            if debug==True:
+                                print ('ocurrence: ' + str(occurrence['_id']) + ' active: ' + str(active['reverse']['expression']))
+                            if translation_legacy_expressions!=[]:
+                                # Do the replacement of the legacy expression
+                                if active['reverse']['expression'] in translation_legacy_expressions[0]:
+                                    active['reverse']['expression'] = translation_legacy_expressions[1][translation_legacy_expressions[0].index(active['reverse']['expression'])]
+                                    if debug==True:
+                                        print(' Replaced expression: "' + active['reverse']['expression'] + '" in object: ' + str(occurrence['_id']))
+                                else:
+                                    print('ERROR: Expression not found in translation file: ' + active['reverse']['expression'] + ' in object: ' + str(occurrence['_id']))
+
+    if 'attributes' in occurrence:
+        for attribute in occurrence['attributes']:
+            if 'expression' in attribute:
+                if re.search(_regex_legacy_expression, attribute['expression']):
+                    if attribute['expression'] not in found_legacy_expressions:
+                        found_legacy_expressions.append(attribute['expression'])
+                    find_object_occurrences.append({'_id':str(occurrence['_id']), 'expression':attribute['expression'], 'type':'attribute.expression', 'service':occurrence['service'], 'subservice':occurrence['subservice'], 'expressionIndex':found_legacy_expressions.index(attribute['expression'])})
+                    if debug==True:
+                        print ('ocurrence: ' + str(occurrence['_id']) + ' attribute: ' + str(attribute['expression']))
+                    if translation_legacy_expressions!=[]:
+                        # Do the replacement of the legacy expression
+                        if attribute['expression'] in translation_legacy_expressions[0]:
+                            attribute['expression'] = translation_legacy_expressions[1][translation_legacy_expressions[0].index(attribute['expression'])]
+                            if debug==True:
+                                print(' Replaced expression: "' + attribute['expression'] + '" in object: ' + str(occurrence['_id']))
+                        else:
+                            print('ERROR: Expression not found in translation file: ' + attribute['expression'] + ' in object: ' + str(occurrence['_id']))
```

Comment:
It seem you use many times the expression `str(occurrence['_id'])`. I'd suggest to define a variable like this at the very beginning:

```
id_str = str(occurrence['_id'])
```

and use it where applicable.